server/src/com/cloud/network/rules/RulesManagerImpl.java

```diff
@@ -1578,8 +1593,13 @@ public PortForwardingRule updatePortForwardingRule(long id, Integer privatePort,
         rule.setState(State.Add);
         if (privatePort != null) {
             rule.setDestinationPortStart(privatePort.intValue());
-            rule.setDestinationPortEnd(privatePort.intValue());
+            rule.setDestinationPortEnd((privateEndPort == null) ? privatePort.intValue() : privateEndPort.intValue());
+        }
+        if (privateEndPort != null) {
+            rule.setDestinationPortStart((privatePort == null) ? privateEndPort.intValue() : privatePort.intValue());
+            rule.setDestinationPortEnd(privateEndPort);
```

Comment:
However, above two can be combine into one "if else" with nested conditons to avoid any duplicate statements